src/compile_src.cpp

```diff
@@ -46,31 +47,24 @@ std::vector<char> src_compiler::compile(const std::vector<src_file>& srcs) const
         fs::path full_path   = td.path / src.path;
         fs::path parent_path = full_path.parent_path();
         fs::create_directories(parent_path);
-        write_buffer(full_path.string(), src.content.data(), src.content.size());
+        write_buffer(full_path, src.content.data(), src.content.size());
         if(src.path.extension().string() == ".cpp")
         {
-            params += " " + src.path.filename().string();
+            params.emplace_back(src.path.filename().string());
             if(out.empty())
                 out = src.path.stem().string() + out_ext;
         }
     }
 
-    params += " -o " + out;
+    params.emplace_back("-o " + out);
 
-    if(not launcher.empty())
-    {
-        td.execute(launcher, compiler + " " + params);
-    }
-    else
-    {
-        td.execute(compiler, params);
-    }
+    td.execute(compiler, params, launcher);
```

Comment:
This should declare the `params` with the compiler if using the launcher:

```cpp
std::vector<std::string> params;
if(not launcher.empty())
    params.push_back(launcher);
params.insert(params.end(), flags.begin(), flags.end());
const std::string& exe = launcher.empty() ? compiler : launcher;
```

Then you can do `td.execute(exe, params)`.